src/Disciplina/DB/DSL/SimpleTxDB.hs

```diff
@@ -0,0 +1,155 @@
+module Disciplina.DB.DSL.SimpleTxDB () where
+
+import Universum
+
+import Codec.Serialise (Serialise(..))
+import Crypto.Error (CryptoFailable(..))
+
+import Disciplina.Crypto.Signing.Class (AbstractPK(..), AbstractSK(..))
+import Disciplina.DB.DSL.Types (QueryTx(..), QueryTxs(..), WHERE(..)
+                               ,TxIdEq(..), TxGrade(..), QueryObj(..)
+                               ,TxsFilterExpr(..), ObjHashEq(..))
+import Disciplina.Educator.Txs (PrivateTxId, PrivateTx(..), EducatorTxMsg(..)
+                               ,StudentTxMsg(..), PrivateTxPayload(..))
+import Disciplina.Crypto (hash, PublicKey, SecretKey)
+import Disciplina.Core (Address(..), CourseId(..), SubjectId, Grade(..)
+                       ,mkAddr, hasPathFromTo
+                       ,activityTypeGraphIndexed)
+import Disciplina.DB.DSL.Interpret (MonadSearchTxObj(..), RunQuery(..))
+import Data.List (union, intersect)
+
+import qualified Crypto.PubKey.Ed25519 as Ed25519
+import qualified Data.ByteString.Char8 as C
+
+
+-- | Simple transaction database
+newtype SimpleTxDB a = SimpleTxDB
+   { getSimpleTxDB :: Reader [PrivateTx] a
+   } deriving (Functor, Applicative, Monad, MonadReader [PrivateTx])
+
+instance MonadSearchTxObj SimpleTxDB where
+  runTxQuery = runSimpleTxQuery
+  runTxsQuery = runSimpleTxsQuery
+  runObjQuery = undefined -- TODO, implement this one, but how are Object defined?
+
+instance Serialise Address where
+  encode (Address h) = encode h
+  decode = Address <$> decode
+
+-- | TODO, where should we put these?
+instance Serialise Grade
+instance Serialise EducatorTxMsg
+instance Serialise StudentTxMsg
+instance Serialise PrivateTxPayload
+instance Serialise CourseId
+instance Serialise PrivateTx
+
+
+-- | Find Tx in db with hash == h
+runSimpleTxQuery :: QueryTx -> SimpleTxDB (Maybe PrivateTx)
+runSimpleTxQuery (SELECTTx _ (TxIdEq (h :: PrivateTxId))) =
+  safeHead.filter ((==h).hash) <$> ask
+
+-- | Find Txs in db with SubjectId == a
+runSimpleTxsQuery :: QueryTxs -> SimpleTxDB [PrivateTx]
+runSimpleTxsQuery (SELECTTxs _ (TxSubjectIdEq sId)) =
+  filter ((==sId).ciSubject._ptxCourseId) <$> ask
+
```

Comment:
I think that it's a bit confusing when you leave a blank line between pattern matches of the same function.